cogeo_mosaic/backends/base.py

```diff
@@ -136,19 +142,59 @@ def update(
 
     def assets_for_tile(self, x: int, y: int, z: int) -> List[str]:
         """Retrieve assets for tile."""
-        return self.get_assets(x, y, z)
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if self.tms.rasterio_crs == mosaic_tms.rasterio_crs:
+            return self.get_assets(x, y, z)
+
+        xmin, ymin, xmax, ymax = self.tms.bounds(x, y, z)
+        return self.assets_for_bbox(
+            xmin,
+            ymin,
+            xmax,
+            ymax,
+            coord_crs=self.tms.rasterio_geographic_crs,
+        )
 
-    def assets_for_point(self, lng: float, lat: float) -> List[str]:
+    def assets_for_point(
+        self,
+        lng: float,
+        lat: float,
+        coord_crs: CRS = WGS84_CRS,
+    ) -> List[str]:
         """Retrieve assets for point."""
-        tile = self.tms.tile(lng, lat, self.quadkey_zoom)
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if coord_crs != mosaic_tms.rasterio_geographic_crs:
+            xs, ys = transform(
+                coord_crs, mosaic_tms.rasterio_geographic_crs, [lng], [lat]
+            )
+            lng, lat = xs[0], ys[0]
+
+        tile = mosaic_tms.tile(lng, lat, self.quadkey_zoom)
+
         return self.get_assets(tile.x, tile.y, tile.z)
 
     def assets_for_bbox(
-        self, xmin: float, ymin: float, xmax: float, ymax: float
+        self,
+        xmin: float,
+        ymin: float,
+        xmax: float,
+        ymax: float,
+        coord_crs: Optional[CRS] = WGS84_CRS,
     ) -> List[str]:
         """Retrieve assets for bbox."""
-        tl_tile = self.tms.tile(xmin, ymax, self.quadkey_zoom)
-        br_tile = self.tms.tile(xmax, ymin, self.quadkey_zoom)
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if coord_crs != mosaic_tms.rasterio_geographic_crs:
+            xmin, ymin, xmax, ymax = transform_bounds(
+                coord_crs,
+                mosaic_tms.rasterio_geographic_crs,
+                xmin,
+                ymin,
+                xmax,
+                ymax,
+            )
```

Comment:
transform the coordinates is Mosaic's geographic coordinates 

TBD: use Mosaic's CRS?

Reply:
maybe I am forgetting the difference between a rasterio and pyproj crs here, but at least with proj it should be safe to directly convert without first going to geographic coords here

Reply:
The question was more, should we first convert to CRS ([`tms._tile`](https://github.com/developmentseed/morecantile/blob/ecabdd1f4a22a8f85e4d8721d03e20d621e64ce6/morecantile/models.py#L717)) instead of Geographic CRS ([`tms.tile`](https://github.com/developmentseed/morecantile/blob/ecabdd1f4a22a8f85e4d8721d03e20d621e64ce6/morecantile/models.py#L769)) later in the code.

Going back and forth from geographic to local can be tricky so I think it will be safer to transform the coordinates to  the tms's CRS (instead of Geographic CRS) and then use `tms._tile` method

Reply:
edited ☝️ 

I think the current implementation is fine and is what we had earlier.